pkg/datagatherer/k8s/dynamic.go

```diff
@@ -150,7 +164,15 @@ func (c *ConfigDynamic) newDataGathererWithClient(ctx context.Context, cl dynami
 		return nil, err
 	}
 	// init shared informer for selected namespaces
-	fieldSelector := generateFieldSelector(c.ExcludeNamespaces)
+	fieldSelector := generateExcludedNamespacesFieldSelector(c.ExcludeNamespaces)
```

Comment:
I gave this function a much more specific name to avoid  any confusion about how it fits in with the user-supplied field-selectors.
Also changed the return type to `fields.Selector` so that the user-supplied selectors can be easily ANDed to it.